unittests/test_api.py

```diff
@@ -191,6 +212,50 @@ def test_get_engagement_not_found(self, mockNewEngagement, mockGet, mockEnv):
         mockNewEngagement.assert_called_once_with(self.product_id)
         response.raise_for_status.assert_called_once()
 
+    @patch('dd_import.environment.Environment')
+    @patch('requests.post')
+    @patch.dict('os.environ', {'DD_URL': 'https://example.com',
+                               'DD_API_KEY': 'api_key',
+                               'DD_ENGAGEMENT_NAME': 'engagement',
+                               'DD_ENGAGEMENT_VERSION': '1.0.1'})
+    def test_new_engagement_with_version(self, mockPost, mockEnv):
+        response = Mock(spec=Response)
+        response.status_code = 200
+        response.text = '{\"id\": 3}'
+        mockPost.return_value = response
+
+        api = Api()
+        id = api.new_engagement(self.product_id)
+
+        self.assertEqual(id, self.engagement_id)
+        today = datetime.date.today().isoformat()
+        url = 'https://example.com/api/v2/engagements/'
+        payload = f'{{"name": "engagement", "product": 2, "target_start": "{today}", "target_end": "2999-12-31", "engagement_type": "CI/CD", "deduplication_on_engagement": false, "status": "In Progress", "version": "1.0.1"}}'
```

Comment:
Does deduplication_on_engagement influence the version?

Reply:
No why?

Reply:
I am not sure what payload does, but there is here "version": "1.0.1" and line 255, there is no version

Reply:
payload is the expected body of the internal rest request. This tool is "configured" using env vars and creates rest requests accordingly. In line 220 you see that the DD_ENGAGEMENT_VERSION is set. That's why the tool includes version in the rest request. DD_ENGAGEMENT_DEDUPLICATION is not set so the request includes the default value. For Version there is no default. If it isn't defined it will not be in the body.